utils/adminPipeline.js

```diff
@@ -298,11 +298,34 @@ export async function getForbiddenPairs(groupId) {
 export async function createForbiddenPair(groupId, forbiddenPair) {
```

Comment:
One of the common tips for development is to make functions do one thing only. With your rewrite, this is no longer true and the name createForbiddenPair is not appropriate.
So either create a dedicated checkIfForbiddenPair function, or rename this one to make it clear what is it supposed to do.